lib/payload/scanner/SSL_TLS/engine.py

```diff
@@ -0,0 +1,674 @@
+        p_len = len(cipher_spec) + len(challenge) + 9
+
+        mask = bin(0x8000) # Most significant bit should be set to indicate no padding
+        bin_plen = bin(p_len)
+
+        packet = ""
+        packet = struct.pack("!H", int(mask,2) + int(bin_plen,2))  # Length of record
+        packet += "\x01" 
+        packet += "\x00\x02"
+        packet += struct.pack("!H", len(cipher_spec))
+        packet += "\x00\x00"
+        packet += "\x00\x10" 
+        packet += cipher_spec
+        packet += challenge
+        return packet
+
+    def tlsHello(self, protocol, cipher = 0):
+        record = "\x16" # Message Type 22
+        record += self.protocols[protocol]
+        cipher_suites = ""
+        if cipher == 0:
+            cipher_list = self.tlsCiphers()
+            for c in cipher_list.keys():
+                cipher_suites += c
+        else:
+            for c in cipher.keys():
+                cipher_suites += c
+        if (len(cipher_suites) % 2) == 0:
+            compression = "\x02\x01\x00"
+            hp_len = len(cipher_suites) + len(compression) + 41
+            record += struct.pack("!H", hp_len)
+            handshake = "\x01" # Client Hello Message
+            h_len = struct.pack("!L", hp_len-4)
+            handshake += h_len[1:]
+            handshake += self.protocols[protocol]
+            handshake += self.__helloRand()
+            handshake += "\x00"
+            handshake += struct.pack("!H", len(cipher_suites))
+            handshake += cipher_suites
+            handshake += compression
+            return record+handshake
+        else:
+            #Error in cipher length!
+            return False
+            #sys.exit(2)
+
+    def doClientHello(self, protocol, cipher = 0):
+        response = False
+        try:
+            if protocol == "SSLv2": # Special case for SSLv2
+                helloMsg = self.ssl2Hello(cipher)
+                self.TCP.sendall(helloMsg)
+                buffer = self.TCP.recv(1024)
+                if buffer:
+                    tmp = bytearray(buffer)
+                    length = hex(tmp[0])[3:]+hex(tmp[1])[2:]
+                    length = int(length, 16) + 2
+                    buffering = True
+                    while buffering:
+                        if len(buffer) == length:
+                            buffering = False
+                            break
+                        more = self.TCP.recv(1024)
+                        if not more:
+                            buffering = False
+                        else:
+                            buffer += more
+                    response = buffer
+            else: # SSLv3 and all TLS versions
+                helloMsg = self.tlsHello(protocol, cipher)
+                self.TCP.sendall(helloMsg)
+                buffer = self.TCP.recv(1024)
+                if buffer:
+                    if not self.contentType(buffer)[0] == 'alert' and not self.contentType(buffer)[1] == 'handshake_failure':
```

Comment:
@pradeepjairamani, this line is longer than 120 characters. Too long according to PEP-8. Please look into it.